spark-bigquery-connector-common/src/test/java/com/google/cloud/spark/bigquery/integration/OpenLineageIntegrationTestBase.java

```diff
@@ -0,0 +1,125 @@
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.junit.rules.ExternalResource;
+
+public class OpenLineageIntegrationTestBase {
+  @ClassRule
+  public static TestDataset testDataset = new SparkBigQueryIntegrationTestBase.TestDataset();
+
+  @ClassRule public static CustomSessionFactory sessionFactory = new CustomSessionFactory();
+
+  protected SparkSession spark;
+  protected String testTable;
+  protected File lineageFile;
+
+  public OpenLineageIntegrationTestBase() {
+    this.spark = sessionFactory.spark;
+    this.lineageFile = sessionFactory.lineageFile;
+  }
+
+  @Before
+  public void createTestTable() {
+    testTable = "test_" + System.nanoTime();
+  }
+
+  protected static class CustomSessionFactory extends ExternalResource {
+    SparkSession spark;
+    File lineageFile;
+
+    @Override
+    protected void before() throws Throwable {
+      lineageFile = File.createTempFile("openlineage_test_" + System.nanoTime(), ".log");
+      lineageFile.deleteOnExit();
+      spark =
+          SparkSession.builder()
+              .master("local")
+              .appName("openlineage_test_bigquery_connector")
+              .config("spark.ui.enabled", "false")
+              .config("spark.default.parallelism", 20)
+              .config("spark.extraListeners", "io.openlineage.spark.agent.OpenLineageSparkListener")
+              .config("spark.openlineage.transport.type", "file")
+              .config("spark.openlineage.transport.location", lineageFile.getAbsolutePath())
+              .getOrCreate();
+      spark.sparkContext().setLogLevel("WARN");
+    }
+  }
+
+  private List<JSONObject> parseEventLogs(File file) throws IOException {
+    Scanner scanner = new Scanner(file);
+    List<JSONObject> eventList = new ArrayList<>();
+    while (scanner.hasNextLine()) {
+      String line = scanner.nextLine();
+      JSONObject event = new JSONObject(line);
+      if (!event.getJSONArray("inputs").isEmpty() && !event.getJSONArray("outputs").isEmpty()) {
+        eventList.add(event);
+      }
+    }
+    return eventList;
+  }
+
+  private String getFieldName(JSONObject event, String field) {
+    JSONObject eventField = (JSONObject) event.getJSONArray(field).get(0);
+    return eventField.getString("name");
+  }
+
+  @Test
+  public void testLineageEvent() {
+    String fullTableName = testDataset.toString() + "." + testTable;
+    Dataset<Row> readDF =
+        spark.read().format("bigquery").option("table", TestConstants.SHAKESPEARE_TABLE).load();
+    readDF.createOrReplaceTempView("words");
+    Dataset<Row> writeDF =
+        spark.sql("SELECT word, SUM(word_count) AS word_count FROM words GROUP BY word");
+    writeDF
+        .write()
+        .format("bigquery")
+        .mode(SaveMode.Append)
+        .option("table", fullTableName)
+        .option("temporaryGcsBucket", TestConstants.TEMPORARY_GCS_BUCKET)
+        .option("writeMethod", "direct")
+        .save();
+    try {
```

Comment:
Since the catch is empty, please add `throws Exception` to the method signature and remove the try-catch

Reply:
done, thanks